src/clua-i-virtual-machine.cpp

```diff
@@ -1048,6 +1058,19 @@ static int machine_obj_index_get_reg_address(lua_State *L) {
     return 1;
 }
 
+/// \brief This is the machine.verify_step() method implementation.
+static int machine_obj_index_verify_step(lua_State *L) {
+    lua_settop(L, 4);
+    cm_hash root_hash_before{};
+    clua_check_cm_hash(L, 1, &root_hash_before);
+    cm_hash root_hash_after{};
+    clua_check_cm_hash(L, 4, &root_hash_after);
+    if (cm_verify_step(&root_hash_before, luaL_checkstring(L, 2), luaL_checkinteger(L, 3), &root_hash_after) != 0) {
```

Comment:
We need this to go through the machine pointer as well. See for example implementation of `cm_verify_step_uarch(const cm_machine *m, const cm_hash *root_hash_before, const char *log, const cm_hash *root_hash_after)`.